### PR TITLE
#19432 enable MiMa check for osgi and cluster-metrics projects

### LINKED ISSUE #19432: enable MiMa for akka-protobuf, akka-osgi, akka-persistence-tck, akka-cluster-metrics

### PR DESCRIPTION
**Pull request by 2m**
Also start checking against 2.4.1 and move appropriate filters for 2.4.1.

Fixes #19432.

### CODE REVIEW

**Comment by drewhk** [comment]
LGTM

File: project/MiMa.scala [hunk 1]
```diff
@@ -571,29 +571,32 @@ object MiMa extends AutoPlugin {
...
+      "2.4.1" -> Seq(
+        // #19008
+        FilterAnyProblem("akka.persistence.journal.inmem.InmemJournal"),
+        FilterAnyProblem("akka.persistence.journal.inmem.InmemStore"),
 
         // #19133 change in internal actor
         ProblemFilters.exclude[MissingMethodProblem]("akka.remote.ReliableDeliverySupervisor.gated"),
 
-        // debug logging in ReplayFilter, change of internal actor
+        // #19200 debug logging in ReplayFilter, change of internal actor
```
Comment (ktoso) [inline project/MiMa.scala]: thanks good

**Comment by ktoso** [comment]
LGTM

**Comment by drewhk** [comment]
merge?

**Comment by drewhk** [comment]
ah, it was not validing yet...

File: project/MiMa.scala [hunk 2]
```diff
@@ -603,9 +606,9 @@ object MiMa extends AutoPlugin {
         ProblemFilters.exclude[MissingMethodProblem]("akka.pattern.BackoffSupervisor.akka$pattern$BackoffSupervisor$$child_="),
         ProblemFilters.exclude[MissingMethodProblem]("akka.pattern.BackoffSupervisor.akka$pattern$BackoffSupervisor$$restartCount"),
         ProblemFilters.exclude[MissingMethodProblem]("akka.pattern.BackoffSupervisor.akka$pattern$BackoffSupervisor$$restartCount_="),
-        ProblemFilters.exclude[MissingMethodProblem]("akka.pattern.BackoffSupervisor.akka$pattern$BackoffSupervisor$$child")
-      ),
-    "2.4.1" -> Seq(
+        ProblemFilters.exclude[MissingMethodProblem]("akka.pattern.BackoffSupervisor.akka$pattern$BackoffSupervisor$$child"),
+
+        // #19487
```
Comment (rkuhn) [inline project/MiMa.scala]: Thanks, I had overlooked this very useful scheme!

**Comment by rkuhn** [comment]
LGTM (apart from the BC failure in validation)

**Comment by patriknw** [comment]
LGTM

**Comment by 2m** [comment]
PLS BUILD

**Pull request merged**